src/components/v2/shared/FundingCycleConfigurationDrawers/NFTDrawer/NFTRewardTierModal.tsx

```diff
@@ -0,0 +1,147 @@
+import { t, Trans } from '@lingui/macro'
+import { Form, Input, Modal } from 'antd'
+import { useForm } from 'antd/lib/form/Form'
+import { ModalMode } from 'components/shared/formItems/formHelpers'
+import { NFTRewardTier } from 'models/v2/nftRewardTier'
+import TooltipLabel from 'components/shared/TooltipLabel'
+import { useEffect } from 'react'
+
+import PaymentThresholdFormItem from './PaymentThresholdFormItem'
+import NFTUpload from './NFTUpload'
+import MaxSupplyFormItem from './MaxSupplyFormItem'
+
+export type NFTFormFields = {
+  paymentThreshold: number
+  maxSupply: number
+  name: string
+  externalLink: string
+  description: string
+  imageUrl: string // IPFS link
+}
+
+export default function NFTRewardTierModal({
+  visible,
+  rewardTiers,
+  setRewardTiers,
+  onClose,
+  mode,
+  editingTierIndex,
+}: {
+  visible: boolean
+  rewardTiers: NFTRewardTier[]
+  setRewardTiers: (rewardTiers: NFTRewardTier[]) => void
+  onClose: VoidFunction
+  isCreate?: boolean
+  mode: ModalMode
+  editingTierIndex?: number
+}) {
+  const [NFTForm] = useForm<NFTFormFields>()
+
+  const onFormSaved = async () => {
+    await NFTForm.validateFields()
+
+    const newTier = {
+      paymentThreshold: parseFloat(NFTForm.getFieldValue('paymentThreshold')),
+      maxSupply: NFTForm.getFieldValue('maxSupply'),
+      imageUrl: NFTForm.getFieldValue('imageUrl'),
+      name: NFTForm.getFieldValue('name'),
+      externalLink: NFTForm.getFieldValue('externalLink'),
+      description: NFTForm.getFieldValue('description'),
+    } as NFTRewardTier
+
+    const newTiers =
+      mode === 'Edit'
+        ? rewardTiers.map((tier, i) =>
+            i === editingTierIndex
+              ? {
+                  ...tier,
+                  ...newTier,
+                }
+              : tier,
+          )
+        : [...rewardTiers, newTier]
+
+    setRewardTiers(newTiers)
+    onClose()
+    if (mode === 'Add') {
+      NFTForm.resetFields()
+    }
+  }
+
+  useEffect(() => {
+    if (editingTierIndex !== undefined) {
+      const rewardTier = rewardTiers[editingTierIndex]
+
+      NFTForm.setFieldsValue({
+        imageUrl: rewardTier.imageUrl,
+        name: rewardTier.name,
+        externalLink: rewardTier.externalLink,
+        description: rewardTier.description,
+        paymentThreshold: rewardTier.paymentThreshold,
+        maxSupply: rewardTier.maxSupply,
+      })
+    }
+  })
+
+  return (
+    <Modal
+      visible={visible}
+      okText={mode === 'Edit' ? t`Save NFT reward` : t`Add NFT reward`}
+      onOk={onFormSaved}
+      onCancel={onClose}
+      title={mode === 'Edit' ? t`Edit NFT reward` : t`Add NFT reward`}
+    >
+      <p>
+        <Trans>
+          Encourage treasury contributions by offering an NFT reward for
+          contributions above a certain threshold.
+        </Trans>
+      </p>
+      <Form layout="vertical" form={NFTForm}>
```

Comment:
**suggestion** as a naming convention, we should probably stick with `nftForm` capitalization.